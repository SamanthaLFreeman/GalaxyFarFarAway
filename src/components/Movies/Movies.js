import React from 'react';
import PropTypes from 'prop-types';
import './Movies.scss';

const Movies = ({film}) => {
  return (
    <section className='movie'>
      <div className="fade"></div>

      <section className="star-wars">

        <div className="crawl">

          <div className="title">
      <p>{film.title}</p>
          </div>

      <p>{film.opening_crawl}</p>  
      <p>{film.release_date}</p>
        </div>

      </section>  
    </section>
  );

}

export default Movies;

Movies.protoTypes = {
  film: PropTypes.object
}