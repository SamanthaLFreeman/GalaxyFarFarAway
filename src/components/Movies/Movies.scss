// .movie {
//   width: 40%;
//   direction: up;
//   scroll-amount: 2; 
//   behavior: scroll;
// }