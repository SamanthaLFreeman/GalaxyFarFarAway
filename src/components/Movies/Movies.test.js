import React from 'react';
import { shallow } from 'enzyme';
import Movies from './Movies';

describe('Movies', () => {
  it('should match the snapshot with all the data passed in correctly including no button has been clicked', () => {
    const data = [
      {
        title: 'Movie',
        opening_crawl: 'blah blah blah',
        release_data: '3019'
      }
    ]
    const wrapper = shallow(<Movies film={data} btnClicked={false}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with all the data passed in correctly including a button has been clicked', () => {
    const data = [
      {
        title: 'Movie',
        opening_crawl: 'blah blah blah',
        release_data: '3019'
      }
    ]
    const wrapper = shallow(<Movies film={data} btnClicked={true} />);
    expect(wrapper).toMatchSnapshot();
  });
});