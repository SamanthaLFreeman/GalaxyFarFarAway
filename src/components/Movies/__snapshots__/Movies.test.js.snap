// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies should match the snapshot with all the data passed in correctly 1`] = `
<section
  className="movie"
>
  <div
    className="fade"
  />
  <section
    className="star-wars"
  >
    <div
      className="crawl"
    >
      <div
        className="title"
      >
        <p />
      </div>
      <p />
      <p />
    </div>
  </section>
</section>
`;
